lib/utils/discover-envs.js

```diff
@@ -0,0 +1,36 @@
+// @ts-check
+const fs = require('fs');
+
+const environmentExtractor = /config\.(.*)\.json/;
+
+/**
+ * @param {string} cwd the directory to search
+ * @returns {string[]} list of environments discovered
+ */
+module.exports = function discoverEnvironments(cwd) {
+    const environments = [];
+    const directoryContents = fs.readdirSync(cwd);
+
+    for (const entry of directoryContents) {
+        const matchResult = environmentExtractor.exec(entry);
+
+        if (matchResult) {
+            environments.push(matchResult[1]);
+        }
+    }
+
+    // Always return in order: production, development, anything else
+    let indexPointer = environments.indexOf('development');
```

Comment:
would it be easier/more readable to use `Array.sort` here? something like https://runkit.com/acburdine/603cc993f37dbb001a14bdbe might work

Reply:
Done, I had 3-4 implementations that didn't work so I eventually decided to brute force it 😅